test/tests/functional/pbs_nodes_json.py

```diff
@@ -91,3 +91,35 @@ def test_check_tab_escape_json(self):
             json.loads("\n".join(n_out))
         except ValueError:
             self.assertFalse(True, "Json failed to load")
+
+    @tags('smoke')
+    def test_validate_pbsnodes_attrs(self):
+        """
+        Test if the pbsnodes outputs all the default attributes.
+        And does not print any new attributes.
+        """
+        cmd = [os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                            'bin', 'pbsnodes')]
+        cmd += ['-v', self.server.shortname, '-Fjson']
+        n_out = self.du.run_cmd(self.server.hostname, cmd=cmd)['out']
+        try:
+            json_object = json.loads("\n".join(n_out))
+        except ValueError:
+            self.assertTrue(False)
+        attrs_pbsnodes_dflts = [
+            'Mom', 'Port', 'pbs_version', 'ntype', 'state', 'pcpus',
+            'resources_available', 'resources_assigned', 'resv_enable',
+            'sharing']
+        node_attr = []
+        self.server.expect(NODE, {'state': 'free'},
+                           id=self.server.shortname, interval=3)
+        for key, val in \
+                json_object['nodes'][self.server.shortname].iteritems():
+            if str(key) not in attrs_pbsnodes_dflts:
+                self.assertFalse(not (key == 'license' and val == 'l'),
```

Comment:
You're testing for three things here.  You want to see the existence of certain attributes and not others except sometimes this one attribute.  I'd make this third case more generic.  There might be more in this third case in the future.  Right now it'll be difficult to add those to the test.  If you make it more generic, then it'll be as easy as adding an attribute to a list.  Make a list of possible default attributes.  Those can appear or not.